### PR TITLE
Update Firehose documentation with dynamic_partitioning_configuration

### PR DESCRIPTION
**Pull request by a-mroz**
Looks like documentation for `dynamic_partitioning_configuration` in the Firehose resource was missed in #20769.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #20769

### CODE REVIEW

File: website/docs/r/kinesis_firehose_delivery_stream.html.markdown [hunk 1]
```diff
@@ -625,6 +626,12 @@ resource "aws_kinesis_firehose_delivery_stream" "example" {
 * `region` - (Optional) If you don't specify an AWS Region, the default is the current region.
 * `version_id` - (Optional) Specifies the table version for the output data schema. Defaults to `LATEST`.
 
+#### dynamic_partitioning_configuration
+
+* `enabled` - (Optional) Enables or disables [dynamic partitioning](https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html). Defaults to `false`.
+* `retry_duration` - (Optional) Total amount of seconds Firehose spends on retries. Valid values between 0 and 7200. Default is 300.
+
+
```
Comment (justinretzolk) [inline website/docs/r/kinesis_firehose_delivery_stream.html.markdown:634]: Removing extra whitespace

```suggestion
```

**Review by justinretzolk — CHANGES_REQUESTED** [review]
Looking at the [AWS Go SDK](https://docs.aws.amazon.com/sdk-for-go/api/service/firehose/#DynamicPartitioningConfiguration) and the [resource definition](https://github.com/hashicorp/terraform-provider-aws/blob/6f623165b909564e959a9b9fce1491be988f7b76/internal/service/firehose/delivery_stream.go#L61-L82), this looks good to me! I had one suggestion (just removing a bit of extra whitespace) but after that, we should be good to get this merged in. Thank you for your contribution! 🚀

**Review by justinretzolk — APPROVED** [review]
Looks good to me! Thank you again for your contribution! I'll get this merged in as soon as the checks finish up. 🚀

**Comment by a-mroz** [comment]
@justinretzolk a reminder. I'm afraid this one was forgotten

**Comment by justinretzolk** [comment]
Hey @a-mroz 👋 Apologies, you're totally correct; I'm not sure how this fell off of my radar, but I appreciate you following up! I'll get this merged now.

**Pull request merged**